### PR TITLE
Don't fail when unknown sample types are encountered

### PR DESCRIPTION
**Pull request by varsill**
This PR:
* allows for parsing samples with unknown type (like `tcmd` samples)
* makes sure, that `stsz` entry list is read only when `sample_size` equals 0 (as defined in the specification)

### CODE REVIEW

File: lib/membrane_mp4/demuxer/isom.ex [hunk 1]
```diff
@@ -639,7 +649,14 @@ defmodule Membrane.MP4.Demuxer.ISOM do
   defp all_pads_connected?(_ctx, %{samples_info: nil}), do: false
 
   defp all_pads_connected?(ctx, state) do
-    tracks = 1..state.samples_info.tracks_number
+    how_many_unsupported_tracks =
+      state.samples_info.sample_tables
+      |> Enum.filter(fn {_track_id, table} ->
+        table.sample_description == nil
+      end)
+      |> length()
```
Comment (FelonEkonom) [inline lib/membrane_mp4/demuxer/isom.ex:657]: You can use https://hexdocs.pm/elixir/1.12/Enum.html#count/2

File: lib/membrane_mp4/demuxer/isom.ex [hunk 2]
```diff
@@ -512,12 +519,14 @@ defmodule Membrane.MP4.Demuxer.ISOM do
   defp sample_description_to_kind(%Membrane.H265{}), do: :video
   defp sample_description_to_kind(%Membrane.AAC{}), do: :audio
   defp sample_description_to_kind(%Membrane.Opus{}), do: :audio
+  defp sample_description_to_kind(_other), do: :unknown
```
Comment (FelonEkonom) [inline lib/membrane_mp4/demuxer/isom.ex:522]: Is the new case used anywhere?

**Review by FelonEkonom — APPROVED** [review]

File: lib/membrane_mp4/demuxer/isom.ex (hunk 2 above)
Comment (FelonEkonom) [reply lib/membrane_mp4/demuxer/isom.ex:522]: If not, let's remove it

File: lib/membrane_mp4/demuxer/isom.ex [hunk 3]
```diff
@@ -430,7 +430,10 @@ defmodule Membrane.MP4.Demuxer.ISOM do
   end
 
   defp match_tracks_with_pads(ctx, state) do
-    sample_tables = state.samples_info.sample_tables
+    sample_tables =
+      state.samples_info.sample_tables
+      |> Enum.reject(fn {_track_id, table} -> table.sample_description == nil end)
+      |> Enum.into(%{})
```
Comment (mat-hek) [inline lib/membrane_mp4/demuxer/isom.ex:436]:
```suggestion
      |> Map.reject(fn {_track_id, table} -> table.sample_description == nil end)
```

Comment (mat-hek) [reply lib/membrane_mp4/demuxer/isom.ex:436]: And let's extract it to a separate function

File: lib/membrane_mp4/demuxer/isom.ex (hunk 2 above)
Comment (varsill) [reply lib/membrane_mp4/demuxer/isom.ex:522]: it's indeed not used, I will remove it

File: lib/membrane_mp4/demuxer/isom.ex [hunk 4]
```diff
@@ -639,7 +645,13 @@ defmodule Membrane.MP4.Demuxer.ISOM do
...
 
   defp all_pads_connected?(ctx, state) do
-    tracks = 1..state.samples_info.tracks_number
+    how_many_unsupported_tracks =
+      state.samples_info.sample_tables
+      |> Enum.count(fn {_track_id, table} ->
+        table.sample_description == nil
+      end)
+
+    tracks = 1..(state.samples_info.tracks_number - how_many_unsupported_tracks)
```
Comment (mat-hek) [inline lib/membrane_mp4/demuxer/isom.ex:654]:
```suggestion
    count_of_supported_tracks =
      state.samples_info.sample_tables
      |> reject_unsupported_sample_types()
      |> Enum.count()

    tracks = 1..count_of_supported_tracks
```

**Review by mat-hek — APPROVED** [review]

**Pull request merged**